### PR TITLE
Aarch64 SHA-512 crash fix

### PR DESCRIPTION
**Pull request by nebeid**
### Description of changes: 
This PR fixes the crash caused by setting the SHA-512 capability bit in `OPENSSL_armcap` when it's not available.
This crash started happening when PR #370 merged changes from BoringSSL, namely https://github.com/awslabs/aws-lc/pull/370/commits/1e15682f1a4bb64c48b84884976a2b5c4201e878, "Enable SHA-512 ARM acceleration when available" and the environment variable `OPENSSL_armcap` has bit 6 manually set, e.g. has value 0x7D.

### Tests
Tested manually on an Aarch64 EC2 instance which doesn't have the SHA-512 capability:
```
$ export OPENSSL_armcap=0x7D && ./crypto/crypto_test
Fatal Error: HW capability found: 0x3D, but HW capability requested: 0x7D.
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c [hunk 1]
```diff
@@ -29,29 +29,45 @@
...
+    sscanf_result = sscanf(in + skip_first_byte + 2, "%" PRIx32, &v);
   } else {
-    sscanf_result = sscanf(in + invert, "%" PRIu32, &v);
+    sscanf_result = sscanf(in + skip_first_byte, "%" PRIu32, &v);
   }
 
   if (!sscanf_result) {
     return;
   }
+  // If getauxval() returned a non-zero hwcap in `armcap` (out)
```
Comment (dkostic) [inline crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:58]: Can we add something like this in the beginning of the comment: `Detect if the user is trying to use the environment variable to set a capability that is _not_ available on the CPU.`?

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c [hunk 2]
```diff
@@ -29,29 +29,45 @@
...
+  // and a bit set in the requested `v` is not set in `armcap`,
+  // abort instead of crashing later.
+  // The case of invert cannot enable an unexisting capability;
+  // it can only disable an existing one.
+  if (!invert && armcap && (~armcap & v))
+  {
+    OPENSSL_PUT_ERROR(SYS, CPU_R_UNSUPPORTED_HW_CAPABILITY);
+    ERR_add_error_dataf("HW capability found: 0x%02x, but HW capability requested: 0x%02x",
+                        armcap, v);
+    // ERR_print_errors_fp(stderr);
```
Comment (dkostic) [inline crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:68]: leftover

**Review by dkostic — COMMENTED** [review]
Should we do the same for x86?

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c [hunk 3]
```diff
@@ -29,29 +29,45 @@
...
+    exit(1);
```
Comment (torben-hansen) [inline crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:69]: If you exit the process how are is setting an error in the error queue useful?

**Comment by nebeid** [comment]
I committed the test I tried and reverted back to keep a record of this attempt.

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c (hunk 1 above)
Comment (nebeid) [reply crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:61]: Done

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c (hunk 2 above)
Comment (nebeid) [reply crypto/fipsmodule/cpucap/cpu_aarch64_linux.c]: Changed to fprintf() as discussed.

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c (hunk 3 above)
Comment (nebeid) [reply crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:71]: Thanks. Changed to fprintf() directly skipping the error queue.

**Review by dkostic — DISMISSED** [review]

**Review by dkostic — APPROVED** [review]

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c [hunk 4]
```diff
@@ -29,30 +29,48 @@
 #include <string.h>
 
 #include <openssl/arm_arch.h>
+#include <openssl/err.h>
```
Comment (torben-hansen) [inline crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:32]: Redundant?

File: crypto/fipsmodule/cpucap/cpu_aarch64_linux.c [hunk 5]
```diff
@@ -29,30 +29,48 @@
...
 
+OPENSSL_DECLARE_ERROR_REASON(SYS, CPU_R_UNSUPPORTED_HW_CAPABILITY);
```
Comment (torben-hansen) [inline crypto/fipsmodule/cpucap/cpu_aarch64_linux.c:34]: Redundant?

File: include/openssl/arm_arch.h [hunk 6]
```diff
@@ -217,4 +217,7 @@
 
 #endif  // __ARMEL__ || _M_ARM || __AARCH64EL__ || _M_ARM64
 
+// Assigning a error value higher than those known in linux system library
+#define CPU_R_UNSUPPORTED_HW_CAPABILITY  200
```
Comment (torben-hansen) [inline include/openssl/arm_arch.h:221]: Redundant?

**Comment by nebeid** [comment]
Removed the leftover redundant error definition. Thanks, Torben.

**Pull request merged**